docs/content/enterprise-addons/ingress.md

```diff
@@ -0,0 +1,194 @@
+# Ingress Addon
+
+A Kubernetes API Server could be announced to users in several ways.
+The most preferred way is leveraging on Load Balancers with their dedicated IP.
+
+However, IPv4 addresses could be limited and scarce in availability, as well as expensive when running in the Cloud.
+A possible optimisation could be implementing an Ingress Controller which routes traffic to Kubernetes API Servers on a host-routing basis.
+
+Despite this solution sounding optimal for end users, it brings some challenges from the worker nodes' standpoint.
+
+## Challenges
+
+Internally deployed applications that need to interact with the Kubernetes API Server will leverage on the `kubernetes` endpoint in the `default` namespace:
+every request sent to the `https://kubernetes.default.svc` endpoint will be forwarded to the Kubernetes API Server.
+
+The routing put in place by the Kubernetes CNI is based on the L4, meaning that all the requests will be forwarded to the Ingress Controller with no `Host` header,
+making impossible a routing based on the FQDN.
+
+## Solution
+
+The `kamaji-addon-ingress` is an addon that will expose the Tenant Control Plane behind an Ingress Controller.
+It's responsible for creating an `Ingress` object with the required HTTP rules, as well as the annotations needed for the TLS/SSL passthrough.
+
+Following is the list of supported Ingress Controllers:
+
+- [HAProxy Technologies Kubernetes Ingress](https://github.com/haproxytech/kubernetes-ingress)
+
+> Active subscribers can request additional Ingress Controller flavours  
+
+## How to enable the Addon
+
+The addon can be enabled for all Tenant Control Plane instances annotated with the key `kamaji.clastix.io/ingress.domain`.
+The value must be the expected suffix domain of generated resources.
+
+```yaml
+apiVersion: kamaji.clastix.io/v1alpha1
+kind: TenantControlPlane
+metadata:
+  annotations:
+    kamaji.clastix.io/ingress.domain: clastix.cloud  # the expected kamaji-addon-ingress label
+  creationTimestamp: "2024-10-03T13:51:29Z"
+  finalizers:
+  - finalizer.kamaji.clastix.io
+  - cloud.clastix.io/paralus
+  - finalizer.kamaji.clastix.io/soot
+  generation: 4
+  labels:
+    clastix.cloud/namespace: default
+    clastix.cloud/user: apezzuto-1003
+    platform: capsule
+  name: tenant-00
+  namespace: apezzuto
+  resourceVersion: "409361036"
+  uid: 592ee7b8-cd07-48cf-b754-76f370c3f87c
+```
+
+Once a Tenant Control Plane has been annotated with this key, the addon will generate the following `Ingress` object.
+
```

Comment:
remove redundand metadata as resourceVersion, generation, etc. as well as custom info as `apezzuto-tenant-00.k8s.clastix.cloud`

Reply:
I'm just keeping the namespace and name for reference and make it easier to understand.